packages/trader/src/Modules/Contract/Containers/__tests__/contract.spec.tsx

```diff
@@ -0,0 +1,79 @@
+import React from 'react';
+import { render, screen } from '@testing-library/react';
+import { Router } from 'react-router-dom';
+import Contract from '../contract';
+import TraderProviders from '../../../../trader-providers';
+import { createMemoryHistory } from 'history';
+
+jest.mock('react-router', () => ({
+    ...jest.requireActual('react-router'),
+    withRouter: (component: React.FC) => component,
+    Redirect: jest.fn(() => 'MockRedirect'),
+}));
+
+jest.mock('react-router-dom', () => ({
+    ...jest.requireActual('react-router-dom'),
+    match: jest.fn(() => 'MockMatch'),
+}));
+
+jest.mock('../contract-replay', () => jest.fn(() => 'MockContractReplay'));
+
+const defaultProps = {
+    match: { params: { contract_id: '123' } },
+};
+
+const MockContract = ({ store, props }: { store: any; props: any }) => {
+    const history = createMemoryHistory();
+    history.push('/contract/123');
+
+    return (
+        <TraderProviders store={store}>
+            <Router history={history}>
+                <Contract {...props} />
+            </Router>
+        </TraderProviders>
+    );
+};
+
+describe('<Contract>', () => {
+    const mockStore = {
+        contract_replay: {
+            has_error: false,
+            error_message: '',
+            error_code: '',
+            removeErrorMessage: jest.fn(),
+            removeAccountSwitcherListener: jest.fn(),
+            setAccountSwitcherListener: jest.fn(),
+        },
+    };
+
+    it('renders ContractReplay component when no error is present', () => {
+        render(<MockContract store={mockStore} props={defaultProps} />);
+
+        expect(screen.getByText('MockContractReplay')).toBeInTheDocument();
+        expect(screen.queryByTestId('dt_error-component')).not.toBeInTheDocument();
+        expect(screen.queryByText('MockRedirect')).not.toBeInTheDocument();
+    });
+
+    it('redirects to 404 page when contract ID is not a number', () => {
+        const props = {
+            match: { params: { contract_id: 'abc' } },
+        };
+
+        render(<MockContract store={mockStore} props={props} />);
+
+        expect(screen.queryByText('MockRedirect')).toBeInTheDocument();
```

Comment:
Hmm, it doesn't work with `Router`, it only works with `BrowserRouter`, but I tried using it and there's another issue appearing related to history since `BrowserRouter` doesn't handle it properly, at least in this case. I think it's fine to just mock redirect in this case. In the end, we're not testing the functioning of redirect here but the logic inside `Contract` which is tested correctly with the mocked Redirect